src/ArrowFunctions.cpp

```diff
@@ -849,6 +876,24 @@ int cpp_readColumnByName(const char* filename, void* chpl_arr, const char* colna
           delete[] def_lvl;
           delete[] rep_lvl;
         }
+      } else if(ty == ARROWDECIMAL) {
+        auto chpl_ptr = (double*)chpl_arr;
+        parquet::FixedLenByteArray value;
+        parquet::FixedLenByteArrayReader* reader =
+          static_cast<parquet::FixedLenByteArrayReader*>(column_reader.get());
+        startIdx -= reader->Skip(startIdx);
+
+        while (reader->HasNext() && i < numElems) {
+          if((numElems - i) < batchSize)
+            batchSize = numElems - i;
+          (void)reader->ReadBatch(1, nullptr, nullptr, &value, &values_read);
+          arrow::Decimal128 v;
+          PARQUET_ASSIGN_OR_THROW(v,
+                                  ::arrow::Decimal128::FromBigEndian(value.ptr, byteLength));
```

Comment:
does it matter if the server's byte order is little endian? I just want to make sure `FromBigEndian` will still work

Reply:
No, the `FromBigEndian` is referring to the Arrow `FixedLenByteArray` type, which is always stored in big endian itself. There isn't a way to convert from a `FixedLenByteArray` type itself to a decimal, so here we have to use the `FromBigEndian` helper function on the byte array pointer, which is always big endian, regardless of machine architecture.